wagtail/wagtailforms/templates/wagtailforms/index_submissions.html

```diff
@@ -23,11 +23,72 @@
                 },
                 lang: 'lang'
             });
+
+            var selectAllCheckbox = document.getElementById('select-all');
+            var deleteButton = document.getElementById('delete-submissions');
+            var selectedSubmissions = {};
+
+            function updateActions() {
+                var everySubmissionSelected = Object.values(selectedSubmissions).every(function(v) { return v; });
+                var someSubmissionsSelected = Object.values(selectedSubmissions).some(function(v) { return v; })
+
+                // Select all box state
+                if (everySubmissionSelected) {
+                    // Every submission has been selected
+                    selectAllCheckbox.checked = true;
+                    selectAllCheckbox.indeterminate = false;
+                } else if (someSubmissionsSelected) {
+                    // At least one, but not all submissions have been selected
+                    selectAllCheckbox.checked = false;
+                    selectAllCheckbox.indeterminate = true;
+                } else {
+                    // No submissions have been selected
+                    selectAllCheckbox.checked = false;
+                    selectAllCheckbox.indeterminate = false;
+                }
+
+                // Delete button state
+                if (someSubmissionsSelected) {
+                    deleteButton.classList.remove('disabled')
+                    deleteButton.style.visibility = "visible";
+                } else {
+                    deleteButton.classList.add('disabled')
+                    deleteButton.style.visibility = "hidden";
+                }
+            }
+
+            // Initialise selectedSubmissions
+            $('input[type=checkbox].select-submission').each(function() {
+                selectedSubmissions[this.dataset.submissionId] = false;
+            });
+
+
+            // Event handlers
+
+            $(selectAllCheckbox).on('change', function() {
+                // Select/deselect all
+                for (var submissionId in selectedSubmissions) {
+                    selectedSubmissions[submissionId] = this.checked;
+                }
+
+                // Update checkbox states in DOM
+                $('input[type=checkbox].select-submission').each(function() {
+                    this.checked = selectedSubmissions[this.dataset.submissionId];
+                });
+
+                updateActions();
+            });
+
+            $('input[type=checkbox].select-submission').on('change', function() {
+                selectedSubmissions[this.dataset.submissionId] = this.checked;
+
+                updateActions();
+            });
         });
     </script>
 {% endblock %}
 {% block content %}
-    <header class="nice-padding">
+    <header class="nice-padding" style="margin-bottom: 0px;">
```

Comment:
Forgot I added this. Thanks for spotting! I'll just leave it the current way for now.